src/nl/codefox/contextvh/sellLand.mod2g

```diff
@@ -5,15 +5,33 @@ exit=always.
 
 % The selling land module
 module sellLand(MultiPoly) {
+	
+	% check if we succeeded in selling the land
+	if bel(not(ourLand(MultiPoly)),offeredLand(MultiPoly,BuyerID,Counter))
+		then delete(offeredLand(MultiPoly,BuyerID,Counter)) + 
+		insert(doneSelling(MultiPoly)) + exit-module.
+	
+	% try another buyer when we have tried it three times with facilities
+	if bel(offeredLand(MultiPoly,BuyerID,Counter),facilitiesID(BuyerID),Counter=3,
```

Comment:
I believe it is better to handle this logic to adopt the goal to sell the correct land instead of checking that in this module. Mostly because it will then only execute the module when there really is a need to sell land.